### PR TITLE
macports.conf: Allow patterns for host matching

### PR DESCRIPTION
**Pull request by raimue**

### CODE REVIEW

File: src/macports1.0/macports.tcl [hunk 1]
```diff
@@ -654,8 +654,7 @@ proc mportinit {{up_ui_options {}} {up_options {}} {up_variations {}}} {
         macports::macosx_deployment_target \
         macports::archivefetch_pubkeys \
         macports::ping_cache \
-        macports::host_blacklisted \
-        macports::host_preferred \
+        macports::host_cache \
```
Comment (neverpanic) [inline src/macports1.0/macports.tcl]: `host_cache` does not seem to be used in mportinit, or did I miss it?

`host_backlist` and `preferred_hosts`, however, are – although I guess technically it was never necessary to add them to this global statement anyway?

File: src/macports1.0/macports.tcl [hunk 2]
```diff
@@ -5219,12 +5213,25 @@ proc macports::revupgrade_buildgraph {port stackname adjlistname revadjlistname
...
-    global macports::ping_cache macports::host_blacklisted macports::host_preferred
-    if {[info exists host_blacklisted($host)]} {
-        return -1
-    } elseif {[info exists host_preferred($host)]} {
-        return 1
-    } elseif {[info exists ping_cache($host)]} {
+    global macports::ping_cache macports::host_cache \
+           macports::host_blacklist macports::preferred_hosts
+
+    if {[info exists host_cache($host)]} {
```
Comment (neverpanic) [inline src/macports1.0/macports.tcl]: Shouldn't there be an `array set macports::host_cache {}` to initialize the array somewhere?

File: src/macports1.0/macports.tcl [hunk 3]
```diff
@@ -5219,12 +5213,25 @@ proc macports::revupgrade_buildgraph {port stackname adjlistname revadjlistname
...
+        return $host_cache($host)
+    }
+    foreach pattern $macports::host_blacklist {
+        if {[string match $pattern $host]} {
```
Comment (neverpanic) [inline src/macports1.0/macports.tcl]: We should probably make this `string match -nocase`.

File: src/macports1.0/macports.tcl [hunk 4]
```diff
@@ -5219,12 +5213,25 @@ proc macports::revupgrade_buildgraph {port stackname adjlistname revadjlistname
...
+            set host_cache($host) -1
+            return -1
+        }
+    }
+    foreach pattern $macports::preferred_hosts {
+        if {[string match $pattern $host]} {
```
Comment (neverpanic) [inline src/macports1.0/macports.tcl]: Same here.

File: src/macports1.0/macports.tcl (hunk 1 above)
Comment (raimue) [reply src/macports1.0/macports.tcl]: Technically not necessary here, but I thought I should list it for completeness.

File: src/macports1.0/macports.tcl (hunk 2 above)
Comment (raimue) [reply src/macports1.0/macports.tcl]: Technically also not necessary, but that would be something for `mportinit`.

File: src/macports1.0/macports.tcl (hunk 3 above)
Comment (raimue) [reply src/macports1.0/macports.tcl]: Sounds reasonable.

**Comment by raimue** [comment]
Updated to address initializing `host_cache` as array and adding `-nocase`.

**Pull request merged**